### DISCUSSION TITLE
Allow ability to select menu choices in OptionButton by Button Press/Release action modes

### DISCUSSION THREAD

**Discussion by yosimba2000** [godotengine/godot-proposals#7981] [GUI]
This is for Godot 4.1 .Net version.

So the OptionButton node itself has a property called Action Mode that allows you to choose when the menu should be displayed: on button press, or on button release. Right now, I have it set to Action Mode = On Button Release. 

When using Mouse as input device, the behavior is as expected. But the behavior is NOT as expected when using controller (I am using PS5 controller).

The expected behavior is this:
1) Click mouse/controller button on the OptionButton.
2) On mouse/controller button release, OptionButton reveals the menu.
3) Click mouse/controller button on any choice in the menu.
4) On release mouse/controller button , the choice is selected, the menu disappears, and Node Signal `item_selected` is emitted.

When using controller, the actual behavior is this:
1) Controller press button on the OptionButton.
2) On Button release, OptionButton reveals the menu.
3) Select any choice in the menu, and press controller button.
4) Right when controller button is pressed, not released, the choice is selected, the menu disappears, and Node Signal `item_selected` is emitted. At this time, the OptionButton is now focused because the choice menu disappeared. Remember, you are still pressing down the controller button.
5) When you RELEASE the controller button, the release is on the OptionButton because the menu disappeared, and therefore the menu reappears again because the OptionButton was set to Action Mode =  On Button Release. 

The same issue occurs with Action Mode =  On Button Press, except it happens when you press the button and not release.

 So as you can see, when using a controller, the user enters a cyclic menu open/close scenario that can only be escaped in one way:

1) When choosing your menu choice, hold down the controller button to select the choice, but do not release button to prevent button release signal on the OptionButton.
2) Select a different GUI node using dpad or whatever control mechanism you have, then you can safely release the button since you are no longer releasing button on the OptionButton.

Solution:
Allow us to select menu options within an OptionButton using Action Mode = On Button Release. So basically the OptionButton itself will keep its Action Mode, but the menu choices inside it will also only be selected using  Action Mode = On Button Release. This way, the menu only closes after user releases controller button, and not on button press.

I have adjusted the default UI_Accept input mapping to also include the X button on the PS5 controller, which is what I'm using to activate the buttons/menus.

**Comment by Sauermann** [comment]
A similar behavior for PopupMenu is reported in https://github.com/godotengine/godot/issues/81297
This might actually be considered a usability bug.

**Reply by yosimba2000** [reply]
hopefully it can be fixed in time for 4.2 release :)
should this thread be labeled as bug?
